src/test/java/net/infordata/em/TerminalClientTest.java

```diff
@@ -290,4 +282,182 @@ public void shouldSendExceptionToExceptionHandlerWhenLoginAndServerDownByLabel()
     exceptionWaiter.awaitException();
   }
 
+  @Test
+  public void shouldGetCorrectFieldsWhenGetFields() throws Exception {
+    awaitKeyboardUnlock();
+   List<TestField> actualFields = client.getFields().stream().map(TestField::new).collect(Collectors.toList());
+    assertEquals(buildExpectedFields(), actualFields);
+  }
+
+  private List<TestField> buildExpectedFields() {
+    List<TestField> testFieldList = new ArrayList<>();
+    testFieldList.add((new TestField(new TestField.FieldBuilder(FIELD_STRING_OUTPUT).withAttr(36).withDefaultColumn().withRow(5))));
+    testFieldList.add((new TestField(new TestField.FieldBuilder(FIELD_STRING_OUTPUT).withAttr(39).withDefaultColumn().withRow(6))));
+    testFieldList.add((new TestField(new TestField.FieldBuilder(FIELD_STRING_OUTPUT).withAttr(36).withDefaultColumn().withRow(7).withFfwModified())));
+    testFieldList.add((new TestField(new TestField.FieldBuilder(FIELD_STRING_OUTPUT).withAttr(36).withDefaultColumn().withRow(8).withFfwModified())));
+    testFieldList.add((new TestField(new TestField.FieldBuilder(FIELD_STRING_OUTPUT).withAttr(36).withDefaultColumn().withRow(9).withFfwModified())));
+  return testFieldList;
+  
+  }
+
+  private static class TestField {
+
+    private int row;
+    private int column;
+    private byte[] ffw;
+    private byte[] fcw;
+    private int attr;
+    private String text;
+    
+    private TestField(FieldBuilder builder) {
+      row = builder.row;
+      column = builder.column;
+      attr = builder.attr;
+      text = builder.text;
+      ffw = builder.ffw;
+      fcw = builder.fcw;
+
+    }
+
+    private TestField(XI5250Field field){
+     column = field.getCol();
+     row = field.getRow();
+     attr = field.getAttr();
+     text = field.getString();
+     ffw = field.getFFW();
+     fcw = field.getFCW();
+    }
+    static final class FieldBuilder {
+
+      private int row;
+
+      private int column;
+      private int attr;
+      private String text;
+      private byte[] ffw = {64, 32};
+      private byte[] fcw = {0, 0};
+      FieldBuilder(String text) {
+        this.text = text;
+      }
+      FieldBuilder withFfwModified() {
+        ffw[0] = 64;
+        ffw[1] = 0; 
+        return this;
+      }
+
+      FieldBuilder withRow(int val) {
+        row = val;
+        return this;
+      }
+
+       FieldBuilder withDefaultColumn() {
+        column = 52;
+        return this;
+      }
+
+       FieldBuilder withAttr(int val) {
```

Comment:
indentation doesn't look consisten, please re format the entire file to avoid such issues.